### PR TITLE
Level aggregator

### PR DESCRIPTION
**Pull request by Tanz0rz**

### CODE REVIEW

File: src/main/java/ludum/dare/levels/LevelWithMeta.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public TileObject leftMostTile;
+    public TileObject rightMostTile;
+
+    public LevelWithMeta(Level level){
+        this.level = level;
+
+        leftMostTile = new TileObject(new BitRectangle(0, 0, 0, 0), false, 0);
+        leftMostTile.rect.xy.x = 999999;
+        rightMostTile = new TileObject(new BitRectangle(0, 0, 0, 0), false, 0);
+        rightMostTile.rect.xy.x = 0;
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/LevelWithMeta.java]: make this -9999999

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        for(Level level : levelsRaw){
+            levels.add(new LevelWithMeta(level));
+        }
+
+        calculateLeftAndRightTilesPerSegment(levels);
+        adjustRectsInEachLevelSegment(levels);
+        buildAggregatedLevels(levels);
+    }
+
+    private void calculateLeftAndRightTilesPerSegment(List<LevelWithMeta> levels){
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java]: Put this method in the LevelWithMeta class

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+ * Created by Admin on 12/12/2015.
+ */
+public class AggregateLevels {
+
+    public static int TILE_SIZE = 16;
+    public static boolean DEBUG = true;
+
+    public AggregateLevels(List<Level> levelsRaw){
+
+        ArrayList<LevelWithMeta> levels = new ArrayList<>();
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java]: This should be List<Level...

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+        for(Level level : levelsRaw){
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java]: levels.forEach(l -> levels.add(new LevelWithMeta(l));

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    private void buildAggregatedLevels(List<LevelWithMeta> levelsWithMetaData){
+        LevelBuilder levelBuilder = new LevelBuilder(TILE_SIZE);
+
+        for(LevelWithMeta level : levelsWithMetaData) {
+
+            for (TileObject[] toa : level.level.gridObjects) {
+                for (TileObject to : toa) {
+                    if (to != null) {
+                        levelBuilder.createLevelObject(new BitPointInt((int) to.rect.xy.x, (int) to.rect.xy.y), new BitPointInt((int) to.rect.xy.x + TILE_SIZE, (int) to.rect.xy.y + TILE_SIZE), false, to.material);
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java]: Take isPassable value from the original tile

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    public static void main(String args[]){
+
+        List<Level> levelsRaw = new ArrayList<Level>();
+
+        String level1Path = "C:\\\\Users\\Admin\\Desktop\\LD34\\LudumDare34\\src\\main\\resources\\levels\\level_1";
+        String level2Path = "C:\\\\Users\\Admin\\Desktop\\LD34\\LudumDare34\\src\\main\\resources\\levels\\level_2";
+        String level3Path = "C:\\\\Users\\Admin\\Desktop\\LD34\\LudumDare34\\src\\main\\resources\\levels\\level_3";
+        String level4Path = "C:\\\\Users\\Admin\\Desktop\\LD34\\LudumDare34\\src\\main\\resources\\levels\\level_4";
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java]: Don't use local absolute paths
From: C:\\Users\Admin\Desktop\LD34\LudumDare34\src\main\resources\levels\level_1
To: src\main\resources\levels\level_1

File: src/main/java/ludum/dare/levels/AggregateLevels.java [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            for (TileObject[] toa : level.level.gridObjects) {
+                for (TileObject to : toa) {
+                    if (to != null) {
+                        levelBuilder.createLevelObject(new BitPointInt((int) to.rect.xy.x, (int) to.rect.xy.y), new BitPointInt((int) to.rect.xy.x + TILE_SIZE, (int) to.rect.xy.y + TILE_SIZE), false, to.material);
+                    }
+                }
+            }
+        }
+
+        LevelUtilities.saveLevel(levelBuilder, false);
```
Comment (Kenoshen) [inline src/main/java/ludum/dare/levels/AggregateLevels.java:120]: This will move, probably

Comment (Tanz0rz) [inline src/main/java/ludum/dare/levels/AggregateLevels.java:120]: COOL

**Pull request merged**